README.md

```diff
@@ -46,7 +56,132 @@ or
 
 `java -jar <location of the jar file>`
 
+## Endpoints
+
+All endpoints require a valid Bearer Access Token passed via the `Authorization` header.
+
+Example request:
+
+```http request
+GET /resources/devices HTTP/1.1
+Authorization: Bearer 5F09FC2DD7DCDB72ABF1A6C026DF8FFB9D7D1F4AD069E34F0A6EC6206A593420
+```
+
+### List devices
+
+GET: `/resources/devices`
+
+Requires an Access token with scope `read`. Access tokens issued via
+the [Implicit Authentication flow](https://docs.onegini.com/msp/stable/token-server/topics/mobile-apps/implicit-authentication/implicit-authentication.html) are
+rejected.
+
+Returns a list of devices for the user for which the Access Token was issued. Refer
+to [Device API v4](https://docs.onegini.com/msp/stable/token-server/api-reference/end-user/device-v4.html) for its response format.
+
+### Details of the application
+
+GET: `/resources/application-details`
+
+Requires an Access token with scope `application-details`. Access tokens issued via the Implicit Authentication flow are rejected.
+
+Returns an object with details of the (mobile) application for which the Access token was issued.
+
+Example response:
+
+```json
+{
+  "application_identifier": "exampleApp",
+  "application_platform": "android",
+  "application_version": "1.0.0"
+}
+```
+
+### Decorated user id
+
+GET: `/resources/user-id-decorated`
+
+Requires an Access token that was issued via the Implicit Authentication flow.
+
+Returns an object with a "decorated" user identifier.
+
+Example response:
+
+```json
+{
+  "decorated_user_id": "✨ myDummyUserId ✨"
+}
+```
+
+### File upload
+
+POST: `/resources/file-upload`
```

Comment:
Is this path inlined with what we have in the controller?

Reply:
No it's not. I'll change the controller and line 132 of the readme